src/embed_tests/Codecs.cs

```diff
@@ -57,31 +66,258 @@ public class Codecs {
         }
 
         [Test]
-        public void TupleRoundtripObject() {
+        public void TupleRoundtripObject()
+        {
             TupleRoundtripObject<ValueTuple<int, string, object>, ValueTuple>();
         }
-        static void TupleRoundtripObject<T, TTuple>() {
+        static void TupleRoundtripObject<T, TTuple>()
+        {
             var tuple = Activator.CreateInstance(typeof(T), 42, "42", new object());
-            using (Py.GIL()) {
+            using (Py.GIL())
+            {
                 var pyTuple = TupleCodec<TTuple>.Instance.TryEncode(tuple);
                 Assert.IsTrue(TupleCodec<TTuple>.Instance.TryDecode(pyTuple, out object restored));
                 Assert.AreEqual(expected: tuple, actual: restored);
             }
         }
 
         [Test]
-        public void TupleRoundtripGeneric() {
+        public void TupleRoundtripGeneric()
+        {
             TupleRoundtripGeneric<ValueTuple<int, string, object>, ValueTuple>();
         }
 
-        static void TupleRoundtripGeneric<T, TTuple>() {
+        static void TupleRoundtripGeneric<T, TTuple>()
+        {
             var tuple = Activator.CreateInstance(typeof(T), 42, "42", new object());
-            using (Py.GIL()) {
+            using (Py.GIL())
+            {
                 var pyTuple = TupleCodec<TTuple>.Instance.TryEncode(tuple);
                 Assert.IsTrue(TupleCodec<TTuple>.Instance.TryDecode(pyTuple, out T restored));
                 Assert.AreEqual(expected: tuple, actual: restored);
             }
         }
+
+        static string GetIntIterableCommands(string instanceName)
+        {
+            var builder = new System.Text.StringBuilder();
+            builder.AppendLine(@"
+class foo():
+    def __init__(self):
+        self.counter = 0
+    def __iter__(self):
+        return self
+    def __next__(self):
+        if self.counter == 3:
+            raise StopIteration
+        self.counter = self.counter + 1
+        return self.counter");
+
+            builder.AppendLine(instanceName + " = foo()");
+            return builder.ToString();
+        }
+
+        static PyObject GetPythonIterable()
+        {
+            var locals = new PyDict();
+            using (Py.GIL())
+            {
+                PythonEngine.Exec(GetIntIterableCommands("foo_instance"), null, locals.Handle);
+            }
+
+            return locals.GetItem("foo_instance");
+        }
```

Comment:
I think you can simply do `Eval("map(lambda x: x, [1,2,3])")`.

Reply:
@lostmsu wow that is clever, nice!